src/fides/api/util/tcf/tcf_experience_contents.py

```diff
@@ -171,21 +175,28 @@ def get_matching_privacy_declarations(db: Session) -> Query:
             System.privacy_policy.label("system_privacy_policy"),
             System.vendor_id,
             PrivacyDeclaration.data_use,
-            PrivacyDeclaration.legal_basis_for_processing,
+            PrivacyDeclaration.overridden_legal_basis_for_processing.label(  # pylint: disable=no-member
+                "legal_basis_for_processing"
+            ),
```

Comment:
I am labeling the "overridden legal basis" as the "legal_basis_for_processing" because at this point, that field combines the original field with the override and contains everything we need. It also meant we needed very little changes downstream.

Reply:
Pros and cons to this though

Reply:
i think i like this, i.e. my intuition is the "earlier" we cut it off the better 👍 but am open to hearing your take on the cons :)

Reply:
Just there’s a lot of logic being encapsulated under this property and then also relabeling it here as an existing attribute can obscure the truth.

But I thought that this fit well with the existing fundamental that nothing shows up in the TCF experience without a legal basis of consent or LI, which made setting it to null fit really nicely with this.  This property covers what would happen if we were using the data map instead to store the override directly. 

Maybe code comments and renaming so the intent is more clear will help.